core/src/main/java/io/grpc/EquivalentAddressGroup.java

```diff
@@ -75,6 +75,9 @@ public int hashCode() {
 
   @Override
   public boolean equals(Object other) {
+    if (other == null) {
```

Comment:
Won't the `instanceof` check handle this case, since `null` is not an instance of `EquivalentAddressGroup`?